src/HDF5_MultiDim.chpl

```diff
@@ -0,0 +1,409 @@
+                remove(filename);
+            }
+            // create the file
+            var file_id: C_HDF5.hid_t = C_HDF5.H5Fcreate(filename.c_str(), C_HDF5.H5F_ACC_TRUNC, C_HDF5.H5P_DEFAULT, C_HDF5.H5P_DEFAULT);
+            defer { // Close file upon exiting scope
+                C_HDF5.H5Fclose(file_id);
+            }
+
+            if file_id < 0 { // Negative file_id means error
+                throw getErrorWithContext(msg="Failure to create/overwrite file, %s".format(filename),
+                                            lineNumber=getLineNumber(), 
+                                            routineName=getRoutineName(), 
+                                            moduleName=getModuleName(), 
+                                            errorClass='FileNotFoundError');
+            }
+
+            // TODO - add the meta data to the file. This is not done yet because current model does not make sense as it is overwritten (or not updated on append)
+        }
+    }
+
+    /*
+    Reads an HDF5 dataset and builds the components as pdarrays
+    The resulting data is always in flattened form (as expected by ArrayView)
+    Adds a pdarray containing the flattened data and a pdarray containing the shape of the object to the symbol table
+    */
+    proc read_hdf_multi_msg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
+        // Currently always load flat as row major
+        var (filename, dset_name) = payload.splitMsgToTuple(2);
+
+        var file_id: C_HDF5.hid_t;
+        var dset_id: C_HDF5.hid_t;
+
+        file_id = C_HDF5.H5Fopen(filename.c_str(), C_HDF5.H5F_ACC_RDONLY, C_HDF5.H5P_DEFAULT);
+
+        if file_id < 0 { // HF5open returns negative value on failure
+            C_HDF5.H5Fclose(file_id);
+            var errorMsg = "Failure accessing file %s.".format(filename);
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);
+        }
+
+        dset_id = C_HDF5.H5Dopen(file_id, dset_name.c_str(), C_HDF5.H5P_DEFAULT);
+        if dset_id < 0 { // HF5open returns negative value on failure
+            C_HDF5.H5Fclose(file_id);
+            var errorMsg = "Failure accessing dataset %s.".format(dset_name);
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);     
+        }
+        
+
+        // check if rank is attr and then get.
+        var rank: int;
+        if C_HDF5.H5Aexists_by_name(dset_id, ".".c_str(), "Rank", C_HDF5.H5P_DEFAULT) > 0 {
+            writeln("Rank attr Found");
+            var rank_id: C_HDF5.hid_t = C_HDF5.H5Aopen_by_name(dset_id, ".".c_str(), "Rank", C_HDF5.H5P_DEFAULT, C_HDF5.H5P_DEFAULT);
+            var attr_type: C_HDF5.hid_t = C_HDF5.H5Aget_type(rank_id);
+            C_HDF5.H5Aread(rank_id, getHDF5Type(int), c_ptrTo(rank));
+            //writeln(rank);
```

Comment:
There are a few places you have `writeln` commented out. Not a big deal, but they should probably be removed or turned into debug logs.

Reply:
updated